# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/mediasink/OutputStreamManager.cpp
# Commit: 85ef72ebde43
# Full Hash: 85ef72ebde439c5de258af43faa9bd83a6e23b26
# Author: Andreas Pehrson <apehrson@mozilla.com>
# Date: 2019-08-29 21:45:25
# Regressor Bug: 1573102
# File Overlap Count: 1
# Description:
#   Bug 1573102 - Remove mGraph from HTMLMediaElement::OutputMediaStream and OutputStreamManager. r=karlt
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D43666
# ==============================================================================

diff -r f61dd73cac3a -r 85ef72ebde43 dom/media/mediasink/OutputStreamManager.cpp
--- a/dom/media/mediasink/OutputStreamManager.cpp	Thu Aug 29 13:32:31 2019 +0000
+++ b/dom/media/mediasink/OutputStreamManager.cpp	Thu Aug 29 13:32:50 2019 +0000
@@ -7,7 +7,7 @@
 #include "OutputStreamManager.h"
 
 #include "DOMMediaStream.h"
-#include "../MediaStreamGraphImpl.h"
+#include "../MediaStreamGraph.h"
 #include "mozilla/dom/MediaStreamTrack.h"
 #include "mozilla/dom/AudioStreamTrack.h"
 #include "mozilla/dom/VideoStreamTrack.h"
@@ -157,16 +157,17 @@
   }
 }
 
-OutputStreamManager::OutputStreamManager(MediaStreamGraphImpl* aGraph,
+OutputStreamManager::OutputStreamManager(MediaStreamGraph* aGraph,
                                          nsIPrincipal* aPrincipal,
                                          AbstractThread* aAbstractMainThread)
     : mAbstractMainThread(aAbstractMainThread),
-      mGraph(aGraph),
+      mDummyStream(aGraph->CreateSourceStream()),
       mPrincipalHandle(
           aAbstractMainThread,
           aPrincipal ? MakePrincipalHandle(aPrincipal) : PRINCIPAL_HANDLE_NONE,
           "OutputStreamManager::mPrincipalHandle (Canonical)") {
   MOZ_ASSERT(NS_IsMainThread());
+  mDummyStream->Suspend();
 }
 
 void OutputStreamManager::Add(DOMMediaStream* aDOMStream) {
@@ -248,7 +249,8 @@
   MOZ_ASSERT(!HasTrackType(aType),
              "Cannot have two tracks of the same type at the same time");
 
-  RefPtr<SourceMediaStream> stream = mGraph->CreateSourceStream();
+  RefPtr<SourceMediaStream> stream =
+      mDummyStream->Graph()->CreateSourceStream();
   if (!mPlaying) {
     stream->Suspend();
   }
@@ -346,7 +348,7 @@
   }
 }
 
-OutputStreamManager::~OutputStreamManager() = default;
+OutputStreamManager::~OutputStreamManager() { mDummyStream->Destroy(); }
 
 #undef LOG
 